### ISSUE TITLE
how to start a new service with my custom provided Dockerfile?

### ISSUE THREAD

**Issue by matifali** [terraform-community-providers/terraform-provider-railway#7]
I want to create a new railway project running a single service from a Dockerfile.
How could this be achieved given the project and service are to be created from the terraform?
I also want to modify the enrtypoint of the Dockerfile.

**Comment by pksunkara** [comment]
Not enough info.

**Comment by matifali** [comment]
Hi @pksunkara, I added more details to see if it helps.

**Comment by pksunkara** [comment]
You use the Railway config file. https://docs.railway.app/deploy/config-as-code

Example:

```
{
  "$schema": "https://railway.app/railway.schema.json",
  "build": {
    "builder": "DOCKERFILE",
    "dockerfilePath": "packages/api/Dockerfile",
    "watchPatterns": ["packages/api"]
  }
}
```

**Comment by matifali** [comment]
But I am looking to achieve this using the terraform provider
I see that we have resources for both service and project.

**Comment by matifali** [comment]
@pksunkara I guess [this](https://github.com/terraform-community-providers/terraform-provider-railway/compare/master...mrbandler:terraform-provider-railway:master) fork is doing exactly this.

**Comment by pksunkara** [comment]
Yes, and I would happily work with anyone (@mrbandler or you) to integrate those changes. After all, open source is about pooling efforts.

**Comment by mrbandler** [comment]
> Yes, and I would happily work with anyone (@mrbandler or you) to integrate those changes. After all, open source is about pooling efforts.

Hi there, @pksunkara & @matifali; I am more than happy to implement changes to align with the current GraphQL API exposed by Railway.

It will just be a while since unfortunately, my company and my position there only allows me to do open-source work in between projects and when I find the time to do it.

Although, I fully intended to do it since I want to use Terraform and Railway for my side projects as well.

If any other culprits need fixing or features to be implemented, please open other issues so I am aware of them.

Many thanks to @pksunkara for the initial bootstrapping of the project!

**Comment by matifali** [comment]
@mrbandler Sounds great. Whenever you find time. It would be nice to have your changes consolidated here. 🌮

**Comment by wertlex** [comment]
Hey!
I added `num_replicas` and `region` here: https://github.com/terraform-community-providers/terraform-provider-railway/pull/27
